### PR TITLE
move ingress defaults from edge/values to greymatter.yaml

### PR DESCRIPTION
**Pull request by chrisbsmith**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

Moves the Edge Ingress default from edge/values.yaml to greymatter.yaml.  Because they are maps, Helm does a merge of the value files.  So instead of greymatter.yaml overwriting the values of edge/values.yaml, Helm would instead merge them.  The Kubernetes API cannot support an ingress type that has mismatched ingress annotations.

For instance, if I tried to apply these values from greymatter.yaml
```yaml
edge:
      sidecar:
        version: 1.0.0
        image: 'docker.production.deciphernow.com/deciphernow/gm-proxy:1.0.0'
        envvars:
          xds_cluster:
            type: value
            value: 'edge'
      edge:
        base_path: /services/edge/1.0.0
        create_edge_secret: true
        ingress:
          apiVersion: extensions/v1beta1
          annotations:
            nginx.ingress.kubernetes.io/ssl-passthrough: "true"
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/backend-protocol: "https"
          # Provide the rule set you want your ingress controller to use
          rules:
            - host: greymatter.development.deciphernow.com
              http:
                paths: 
                  - path: /
                    backend:
                      serviceName: edge
                      servicePort: 8080
```

I would get this response from the helm template of the edge-ingress
```yaml
➜  decipher-development git:(master) ✗ aws-vault exec decipher-dev -- kubectl get ingress -n greymatter -o yaml
apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      flux.weave.works/antecedent: greymatter:helmrelease/greymatter
      ingress.appscode.com/ssl-passthrough: "true"
      ingress.appscode.com/type: NodePort
      kubernetes.io/ingress.class: voyager
      nginx.ingress.kubernetes.io/backend-protocol: https
      nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
      nginx.ingress.kubernetes.io/ssl-passthrough: "true"
    creationTimestamp: "2019-12-09T19:18:51Z"
    generation: 1
    name: edge
    namespace: greymatter
    resourceVersion: "6346094"
    selfLink: /apis/extensions/v1beta1/namespaces/greymatter/ingresses/edge
    uid: bbdb696b-1ab8-11ea-a97e-160ffdcfc4d5
  spec:
    rules:
    - host: greymatter.development.deciphernow.com
      http:
        paths:
        - backend:
            serviceName: edge
            servicePort: 8080
          path: /
  status:
    loadBalancer: {}
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

Notice that voyager and nginx annotations exist

2. What **changes to custom.yaml** are required?

the edge ingress options were moved to greymatter.yaml

3. Have you documented any additional setup steps or configurations options?

n/a

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Templating matches the same as the current deployment

### CODE REVIEW

File: edge/values.yaml [hunk 1]
```diff
@@ -47,52 +47,18 @@ edge:
...
   ingress:
-    apiVersion: voyager.appscode.com/v1beta1
+    # For ingress, you can specify which API verison  to use
+    apiVersion: extension/v1beta1
   
     # Specific annotations for your deployment
     # Choosing the ingress.appscode.com/type is important and depends on your environment type.
     #   - If using EKS, you need to configure the Type as LoadBalancder 
     #   - If using voyager, you need to configure the type as NodePort
     annotations:
```
Comment (kaitmore) [inline edge/values.yaml]: Defaulting to an empty map here will get rid of console warnings
```suggestion
    annotations: {}
```

File: edge/values.yaml [hunk 2]
```diff
@@ -47,52 +47,18 @@ edge:
...
-          backend:
-            serviceName: edge
-            servicePort: 8080
-      - tcp:
-          port: '443'
-          nodePort: "30000"
-          backend:
-            serviceName: edge
-            servicePort: 8080
+    rules:
```
Comment (kaitmore) [inline edge/values.yaml]: Same here, except it needs []
```suggestion
    rules: []
```

Comment (chrisbsmith) [reply edge/values.yaml]: Good catch!  thanks

Comment (chrisbsmith) [reply edge/values.yaml]: I've pushed an update for this.

**Comment by kaitmore** [comment]
@chrisbsmith One more thing @tilleryd made me realize - we are not using `k8s_use_voyager_ingress` anymore - should we fix that here, or would you prefer a new PR?

**Pull request merged**